### PR TITLE
data renders on page in roughly accurate positions

### PR DESCRIPTION
**Pull request by michchen**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    this.getData = function(prodNum, appObj) {
+      prodNum = prodNum || 1;
+
+      var reqProduct = $.get({
+        url: "http://localhost:9001/get",
+        context: this,
+        data: { id: prodNum }
+      });
+      reqProduct.done(res => {
+        this.setState(res[0]);
```
Comment (nr431207) [inline client/src/App.jsx]: is this the right format for setState? typically Ive seen setState({key: value})

File: database/faker.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const db = require('./db');
+
+const colors_arr = ["Medium Spring Green","Coral","Lawn Green","Yellow","Orange","Light Steel Blue","Fire Brick","Light Grey","Dark Goldenrod","Burly Wood","Dark Slate Blue","Cornflower Blue","Powder Blue","Dark Blue","Dark Slate Gray","Maroon","Silver","Light Salmon","Seashell","Medium Sea Green"];
+
+const randImArr = function (){
+  let num = Math.round(Math.random()*3)+1;
+  return "[" + cats.data.sort(()=>0.5-Math.random()).slice(0,num).map(x=>`'${x}'`).join(',') + "]";
+}
+
+const createQuery = function(){
```
Comment (nr431207) [inline database/faker.js]: Nice job on this function!

File: client/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    this.getData = function(prodNum, appObj) {
+      prodNum = prodNum || 1;
+
+      var reqProduct = $.get({
+        url: "http://localhost:9001/get",
```
Comment (avincenthill) [inline client/src/App.jsx]: use env vars here

File: client/src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        context: this,
+        data: { id: prodNum }
+      });
+      reqProduct.done(res => {
+        this.setState(res[0]);
+        window.state = this.state;
+
+        var reqVariations = $.get({
+          url: "http://localhost:9001/var",
```
Comment (avincenthill) [inline client/src/App.jsx]: env

File: client/src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    };
+
+    this.addImages = function(images) {
+      if (this.state.variations) {
+        console.log(this.state);
+        this.setState({ images: images });
+        window.images = this.state.images
+        let result = {};
+
+        for (var i = 0; i < images.length; i++) {
```
Comment (avincenthill) [inline client/src/App.jsx]: prefer forEach loops to for loops

File: client/src/components/ProductInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
+const React = require("react");
+import DescBullet from "./DescBullet.jsx";
+
+const addComma = function(num) {
+  return String(num).replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (avincenthill) [inline client/src/components/ProductInfo.jsx]: impressive use of regex, not sure what for tho, perhaps comment when you're doing stuff like this

File: client/src/components/ProductInfo.jsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+const ProductInfo = props => (
+  <div>
+    {props.data.category_name ? (
```
Comment (avincenthill) [inline client/src/components/ProductInfo.jsx]: destructure props to avoid writing it a bunch here and below

File: database/faker.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+const cats = require('./cats');
+const faker = require('faker');
+const db = require('./db');
+
+const colors_arr = ["Medium Spring Green","Coral","Lawn Green","Yellow","Orange","Light Steel Blue","Fire Brick","Light Grey","Dark Goldenrod","Burly Wood","Dark Slate Blue","Cornflower Blue","Powder Blue","Dark Blue","Dark Slate Gray","Maroon","Silver","Light Salmon","Seashell","Medium Sea Green"];
+
+const randImArr = function (){
+  let num = Math.round(Math.random()*3)+1;
+  return "[" + cats.data.sort(()=>0.5-Math.random()).slice(0,num).map(x=>`'${x}'`).join(',') + "]";
```
Comment (avincenthill) [inline database/faker.js]: prefer comment what this this doing

**Pull request closed without merging**